From f714d709716d593a454cf38d90218ed7da298de5 Mon Sep 17 00:00:00 2001
From: Rong Dian <b38775@freescale.com>
Date: Fri, 27 Apr 2012 15:48:59 +0800
Subject: [PATCH 1379/2463] ENGR00181094-1:  MX6 SABRESD: Add pin config for
 MAX8903

Configure PINMUX for max8903 driver on SABRESD_rev.B board.

Signed-off-by: Rong Dian <b38775@freescale.com>
---
 arch/arm/configs/imx6_defconfig        |    3 +-
 arch/arm/mach-mx6/board-mx6q_sabresd.c |   34 +++++++++++++++++++++++++++++++-
 2 files changed, 35 insertions(+), 2 deletions(-)

diff --git a/arch/arm/configs/imx6_defconfig b/arch/arm/configs/imx6_defconfig
index b884f39..70be820 100644
--- a/arch/arm/configs/imx6_defconfig
+++ b/arch/arm/configs/imx6_defconfig
@@ -1285,7 +1285,8 @@ CONFIG_POWER_SUPPLY=y
 # CONFIG_BATTERY_MAX17040 is not set
 # CONFIG_BATTERY_MAX17042 is not set
 # CONFIG_CHARGER_ISP1704 is not set
-CONFIG_CHARGER_MAX8903=y
+# CONFIG_CHARGER_MAX8903 is not set
+CONFIG_BATTERY_MAX8903=y
 # CONFIG_CHARGER_GPIO is not set
 CONFIG_HWMON=y
 # CONFIG_HWMON_VID is not set
diff --git a/arch/arm/mach-mx6/board-mx6q_sabresd.c b/arch/arm/mach-mx6/board-mx6q_sabresd.c
index 1c3545c..b7e48c7 100644
--- a/arch/arm/mach-mx6/board-mx6q_sabresd.c
+++ b/arch/arm/mach-mx6/board-mx6q_sabresd.c
@@ -46,7 +46,7 @@
 #include <linux/memblock.h>
 #include <linux/gpio.h>
 #include <linux/etherdevice.h>
-#include <linux/power/max8903_charger.h>
+#include <linux/power/max8903_battery.h>
 #include <linux/regulator/anatop-regulator.h>
 #include <linux/regulator/consumer.h>
 #include <linux/regulator/machine.h>
@@ -1627,6 +1627,38 @@ static void __init mx6_sabresd_board_init(void)
 		gpio_direction_input(SABRESD_PFUZE_INT);
 		mx6q_sabresd_init_pfuze100(SABRESD_PFUZE_INT);
 	}
+	/* set DOK gpio input */
+	ret = gpio_request(SABRESD_CHARGE_DOK_B, "max8903-DOK");
+	if (ret) {
+		printk(KERN_ERR"request max8903-DOK error!!\n");
+		return;
+	} else {
+		gpio_direction_input(SABRESD_CHARGE_DOK_B);
+	}
+	/* set UOK gpio input */
+	ret = gpio_request(SABRESD_CHARGE_UOK_B, "max8903-UOK");
+	if (ret) {
+		printk(KERN_ERR"request max8903-UOK error!!\n");
+		return;
+	} else {
+		gpio_direction_input(SABRESD_CHARGE_UOK_B);
+	}
+	/* set FLT gpio input */
+	ret = gpio_request(SABRESD_CHARGE_FLT_1_B, "max8903-FLT");
+	if (ret) {
+		printk(KERN_ERR"request max8903-FLT error!!\n");
+		return;
+	} else {
+		gpio_direction_input(SABRESD_CHARGE_FLT_1_B);
+	}
+	/* set CHG gpio input */
+	ret = gpio_request(SABRESD_CHARGE_CHG_1_B, "max8903-CHG");
+	if (ret) {
+		printk(KERN_ERR"request max8903-CHG error!!\n");
+		return;
+	} else {
+		gpio_direction_input(SABRESD_CHARGE_CHG_1_B);
+	}
 	/* SPI */
 	imx6q_add_ecspi(0, &mx6q_sabresd_spi_data);
 	spi_device_init();
-- 
1.7.7.4

